server/events/identify.js

```diff
@@ -15,12 +15,18 @@ const IGNORED_TRAITS = [
   "visitToken"
 ];
 
-function updateUser(hull, user) {
+function updateUser(hull, user, shipSettings) {
   try {
     const { userId, anonymousId, traits = {} } = user;
-    if (!userId && !anonymousId) { return false; }
+    const { email } = traits || {};
 
-    return scoped(hull, user).traits(traits).then(
+    if (shipSettings.ignore_segment_userId === true && !email && !anonymousId) {
+      return false;
```

Comment:
@SMK1085 it looks like we already missed that part in logging before, but now it may be a good time to fix it.
In both places we should have `incoming.user.skip`, right?